pyproject.toml

```
@@ -50,5 +55,24 @@ local_scheme = "no-local-version"
 
 [tool.black]
 
+[tool.ruff]
+select = [
+  "E", # pycodestyle errors
+  "F", # pyflakes errors
+  "I", # isort
+  "ISC", # flake8-implicit-str-concat
+  "PGH", # pygrep-hooks
+  "RUF100", # unused noqa (yesqa)
+  "UP", # pyupgrade
+  "W", # pycodestyle warnings
+  "YTT", # flake8-2020
+]
+extend-ignore = [
+  "E203", # Whitespace before ':'
+  "E221", # Multiple spaces before operator
+  "E226", # Missing whitespace around arithmetic operator
+  "E241", # Multiple spaces after ','
```

```suggestion
  "E241", # Multiple spaces after ','
  "UP038", # makes code slower and more verbose
```

https://github.com/astral-sh/ruff/issues/7871